prometheus/src/main/scala/filodb/prometheus/ast/Expressions.scala

```diff
@@ -95,26 +96,35 @@ trait Expressions extends Aggregates with Functions {
 
           // node_info + http_requests
           case (lh: PeriodicSeries, rh: PeriodicSeries) =>
-            val seriesPlanLhs = lh.toSeriesPlan(timeParams)
-            val seriesPlanRhs = rh.toSeriesPlan(timeParams)
-            val cardinality = if (operator.getPlanOperator.isInstanceOf[SetOperator])
-              Cardinality.ManyToMany
-            else
-              vectorMatch.map(_.cardinality.cardinality).getOrElse(Cardinality.OneToOne)
-
-            val matcher = vectorMatch.flatMap(_.matching)
-            val onLabels = matcher.filter(_.isInstanceOf[On]).map(_.labels)
-            val ignoringLabels = matcher.filter(_.isInstanceOf[Ignoring]).map(_.labels)
-
-            BinaryJoin(seriesPlanLhs, operator.getPlanOperator, cardinality, seriesPlanRhs,
-              onLabels.getOrElse(Nil), ignoringLabels.getOrElse(Nil),
-              vectorMatch.flatMap(_.grouping).map(_.labels).getOrElse(Nil))
-
+            //10/2 + foo
+            if (hasScalarResult(lh)) {
+              val scalar = lh.toSeriesPlan(timeParams).asInstanceOf[ScalarPlan]
+              val seriesPlan = rh.toSeriesPlan(timeParams)
+              ScalarVectorBinaryOperation(operator.getPlanOperator, scalar, seriesPlan, scalarIsLhs = true)
```

Comment:
if both LHS and RHS are scalar, it come here. Wouldn't it be wrong ? It should be a scalar/vector operation

Reply:
If both LHS and RHS are scalars it will go to the if block above(starting from line 45) and it will be converted to ScalarBinaryOperation

Reply:
Ack. Please ensure we have a test case to verify that.

Reply:
Yes we have 4 test cases in ScalarQueriesSpec